frontend/src/test/scala/bloop/RunSpec.scala

```diff
@@ -469,4 +469,32 @@ class RunSpec extends BloopHelpers {
     }
   }
 
+  @Test
+  def runUsesRuntimeEnvironment(): Unit = {
+    TestUtil.withinWorkspace { workspace =>
+      object Sources {
+        val `a/A.scala` =
+          """/a/A.scala
+            |object A {
+            |  def main(args: Array[String]): Unit = {
+            |    assert("goodbye" == sys.props("test"))
```

Comment:
It might be tricky to implement, but how hard would it be to assert that the runtime property is not available at compile-time?

Reply:
It would be difficult to test, because we fork compilation only for Java, and under certain conditions only (ie. the Java home is different from the current Java). I guess we'd need a Java annotation processor that would check for the property?